samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -752,16 +752,79 @@ static int init(struct config_options *options)
 	return err;
 }
 
-static void update_config(const u8_t config_id, const u8_t *data, size_t size)
+static inline bool is_my_config_id(u8_t config_id)
+{
+	return (GROUP_FIELD_GET(config_id) == EVENT_GROUP_SETUP) &&
+	       (MOD_FIELD_GET(config_id) == SETUP_MODULE_SENSOR);
+}
+
+static size_t get_config_data_size(u8_t config_id)
 {
 	struct config_options *options;
+	size_t size;
 
-	if ((GROUP_FIELD_GET(config_id) != EVENT_GROUP_SETUP) ||
-	    (MOD_FIELD_GET(config_id) != SETUP_MODULE_SENSOR)) {
-		/* Not for us */
-		return;
+	switch (OPT_FIELD_GET(config_id)) {
+	case SENSOR_OPT_CPI:
+		size = sizeof(options->cpi);
+		break;
+
+	case SENSOR_OPT_DOWNSHIFT_RUN:
+		size = sizeof(options->time_run);
+		break;
+
+	case SENSOR_OPT_DOWNSHIFT_REST1:
+		size = sizeof(options->time_rest1);
+		break;
+
+	case SENSOR_OPT_DOWNSHIFT_REST2:
+		size = sizeof(options->time_rest2);
+		break;
+
+	default:
+		LOG_WRN("Unsupported sensor option (%" PRIu8 ")", config_id);
+	}
+
+	return size;
+}
+
+static void fetch_config(u8_t config_id, u8_t *data)
+{
+	u32_t val;
+
+	switch (OPT_FIELD_GET(config_id)) {
+	case SENSOR_OPT_CPI:
+		val = atomic_get(&sensor_cpi);
+		sys_put_le32(val, data);
+		break;
+
+	case SENSOR_OPT_DOWNSHIFT_RUN:
+		val = atomic_get(&sensor_downshift_run);
+		sys_put_le32(val, data);
+		break;
+
+	case SENSOR_OPT_DOWNSHIFT_REST1:
+		val = atomic_get(&sensor_downshift_rest1);
+		sys_put_le32(val, data);
+		break;
+
+	case SENSOR_OPT_DOWNSHIFT_REST2:
+		val = atomic_get(&sensor_downshift_rest2);
+		sys_put_le32(val, data);
+		break;
+
+	default:
+		goto error;
 	}
 
+	return;
+error:
+	LOG_WRN("Unsupported sensor option (%" PRIu8 ")", config_id);
```

Comment:
this log could be removed/changed to assert - we shouldn't enter this function is this is not for us